### PR TITLE
Update CI for SGX SDK action update

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Also update the rust toolchain action to https://github.com/dtolnay/rust-toolchain

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #14** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/14" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-std/14?utm_source=stack-comment).

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -13,30 +13,24 @@ env:
 
 jobs:
   rustfmt:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly-2022-07-22
```
Comment (jcape) [inline .github/workflows/ci.yaml]: Suggest this be `@master` with the rust toolchains listed in a strategy matrix.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -54,130 +48,83 @@ jobs:
...
   clippy:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     needs:
       - "rustfmt"
       - "markdown-lint"
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly-2022-07-22
```
Comment (jcape) [inline .github/workflows/ci.yaml]: Same comment (use strategy matrix)

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -54,130 +48,83 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
       - uses: mobilecoinfoundation/actions/sgxsdk@main
-      - uses: actions-rs/toolchain@v1
         with:
-          profile: minimal
-          toolchain: ${{ matrix.rust }}
-          override: true
+          version: 2.18.100.3
+      - uses: dtolnay/rust-toolchain@nightly-2022-07-22
```
Comment (jcape) [inline .github/workflows/ci.yaml]: Same comment (use strategy matrix)

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -54,130 +48,83 @@ jobs:
...
```
Comment (jcape) [inline .github/workflows/ci.yaml]: Same comment

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -54,130 +48,83 @@ jobs:
...
-        rust:
-          - nightly-2022-07-22
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ matrix.rust }}
-          override: true
+      - uses: dtolnay/rust-toolchain@nightly-2022-07-22
```
Comment (jcape) [inline .github/workflows/ci.yaml]: Same comment

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -54,130 +48,83 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
       - uses: mobilecoinfoundation/actions/sgxsdk@main
-      - uses: actions-rs/toolchain@v1
         with:
-          profile: minimal
-          toolchain: nightly-2022-07-22
-          override: true
+          version: 2.18.100.3
+      - uses: dtolnay/rust-toolchain@nightly-2022-07-22
```
Comment (jcape) [inline .github/workflows/ci.yaml]: Same comment (rust toolchain from strategy matrix)

**Review by jcape — APPROVED** [review]
Mostly LGTM, but we should always use `rust-toolchain@master` with a strategy matrix, because:
1. I don't think there are tags for each nightly
2. The `rust-toolchain` action doesn't change the default, so even `rust-toolchain@nightly` (which does exist) requires one to also do `cargo +nightly foo` at every invocation.

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -54,130 +49,93 @@ jobs:
...
         with:
-          toolchain: stable
-          profile: minimal
-      - uses: actions-rs/install@v0.1
-        with:
-          crate: cargo-sort
-          version: latest
-          use-tool-cache: true
-      # We run this manually because actions-rs/cargo doesn't have output redirect
+          toolchain: nightly-2022-07-22
```
Comment (jcape) [inline .github/workflows/ci.yaml:60]: We probably want the `stable` toolchain for this.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (nick-mobilecoin) [reply .github/workflows/ci.yaml]: used master with matrix strategy

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (nick-mobilecoin) [reply .github/workflows/ci.yaml]: used @master with matrix strategy

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (nick-mobilecoin) [reply .github/workflows/ci.yaml]: used `@master` with matrix strategy

File: .github/workflows/ci.yaml (hunk 6 above)
Comment (nick-mobilecoin) [reply .github/workflows/ci.yaml]: used `@master` with matrix strategy

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (nick-mobilecoin) [reply .github/workflows/ci.yaml]: used `@master` with matrix strategy

File: .github/workflows/ci.yaml (hunk 7 above)
Comment (nick-mobilecoin) [reply .github/workflows/ci.yaml]: removed the toolchain action from both. By default the images more or less track stable.

**Review by awygle — APPROVED** [review]

**Pull request merged**